test/common/grpc/grpc_client_integration_test.cc

```diff
@@ -254,6 +254,7 @@ TEST_P(GrpcClientIntegrationTest, ServerInitialMetadata) {
   const TestMetadata initial_metadata = {
       {Http::LowerCaseString("foo"), "bar"},
       {Http::LowerCaseString("baz"), "blah"},
+      {Http::LowerCaseString("binary-bin"), "help"},
```

Comment:
If you use `expectInitialMetadata`, I guess you can test it is base64 encoded?

Reply:
Yeah, I just want to make sure it covers the addition first. I am not sure we can intercept headers in transport with gRPC.

Reply:
OK, this produces the right coverage. Do you know how to access h2 headers in transport with grpc lib? That would be the right way to validate base64 encoding.

Reply:
`sendServerInitialMetadata` does send over this to fake_stream so this is already going through gRPC lib, and it actually calls `expectServerInitialMetadata`, can you verify this test fails before your change? basically this `binary-bin` is 1) sent to fake_stream 2) grpc lib does base64 decode 3) metadataTranslate does base64encode, so the `expectInitialMetadata` works.

Reply:
Confirming that this test fails without the change to source:
```
Expected equality of these values:
  value.second
    Which is: "help"
  stream_headers.get_(value.first)
    Which is: "\x85\xE9i"
```